plugins/github/komand_github/connection/connection.py

```diff
@@ -15,31 +18,38 @@ def __init__(self):
         self.username = ""
         self.api_prefix = "https://api.github.com"
         self.secret = None
-        self.github_session = None
 
     def connect(self, params={}):
         self.logger.info("Connect: Connecting..")
         try:
-            self.username = params.get("credentials").get("username")
-            self.secret = params.get("credentials").get("password")
-            self.basic_auth = (self.username, self.secret)
-            self.github_session = requests.Session()
-            self.github_user = github.Github(self.username, self.secret)
-            user_info = self.github_user.get_user()
+            credentials = params.get(Input.CREDENTIALS)
+            self.username = credentials.get("username")
+            self.secret = credentials.get("personal_token", {}).get("secretKey")
+            self.auth_header = {"Authorization": f"Bearer {self.secret}"}
+            self.github_user = github.Github(auth=github.Auth.Token(self.secret))
             self.user = self.github_user.get_user(self.username)
-            self.github_session_user = requests.get(self.api_prefix, auth=(self.username, self.secret), verify=True)
-            if str(self.github_session_user.status_code).startswith("2"):
-                self.logger.info("Connect: Login successful")
-            else:
-                self.logger.info("Connect: Login unsuccessful")
-
-        except github.GithubException as err:
-            self.logger.error("Github: Connect: error %s", err.data)
-            raise Exception("Github: Connect: user could not be authenticated please try again.")
+            self.github_session = requests.Session()
 
-        except requests.exceptions.RequestException as e:
-            raise e
+        except github.GithubException:
+            raise PluginException(
+                cause="Github: Connect: user could not be authenticated please try again.",
+                assistance="Please check that your username and personal token is correct.",
```

Comment:
Can we pass back data here to help debug if suitable?

Reply:
I have added all generic error handling to the util file and passing back data